lxd/storage/backend_lxd.go

```diff
@@ -3163,10 +3163,15 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 		}
 	}
 
-	// Validate config and create database entry for new storage volume.
-	err = VolumeDBCreate(b, project.Default, fingerprint, "", drivers.VolumeTypeImage, false, volConfig, time.Now().UTC(), time.Time{}, contentType, false)
-	if err != nil {
-		return err
+	// For drivers which support both block mode and non-block mode, there should always only be a
+	// single database entry even if two different volumes exist on disk.
+	// We therefore check for an existing entry, and only create one if not present.
+	if imgDBVol == nil {
```

Comment:
I still dont see why we need this, we should only get to this part if the record doesn't already exist, see https://github.com/lxc/lxd/blob/2a1524aedc2962ce56c01482cca194a099480be4/lxd/storage/backend_lxd.go#L3128

What errors were you getting that required this?

Reply:
I just ran the test again, and it doesn't show an error anymore. Another fix probably solved this. I'll remove it.